### ISSUE TITLE
forAllValuesIfExists doesn't pass for env value `null`

### ISSUE THREAD

**Issue by broom9** [bluebirds-blue-jay/access-control#28]
forAllValuesIfExists doc says:
> This modifier does exactly the same thing as forAllValues expect that it ignores null/undefined values.

But it actually only recognizes `undefined` as non-exist value, not `null`. It passes for this case:
```
const condition = {
  stringEquals: {
    forAllValuesIfExists: {
      foo: ['bar', 'baz', 'boo']
    }
  }
};


const ok = {
  foo: undefined
}

const notOk = {
  foo: null // this doesn't pass based on my testing. I think it's a bug
}
```

The "condition-modifiers-manager.js" has a function:
```
    exists(value) {
        return !Lodash.isUndefined(value);
    }
```

It seems only `undefined` is recognized as "exists".

**Comment by SylvainEstevez** [comment]
Looks like there is a clear discrepancy between the docs and the actual behavior, sorry for that @broom9, that's confusing.

I'm not fundamentally against counting `null` as part of the `notExists` concept, but I first need to check what the implications would be in my own projects as I generally tend to consider "the absence of value" `null` as an explicit value - it that makes any sense - meaning that a key with value `null` is indeed considered to exist.

A solution could be to count `null` values as you describe based on a configuration flag. I'll get back to you about this shortly.

**Comment by broom9** [comment]
Thanks. In JS, I agree `null` and `undefined` have different meanings.

In our case, a value is read from a database. And a database doesn't really distinguish `null` vs `undefined` so the undefined value is read as `null`.

**Comment by SylvainEstevez** [comment]
@broom9 I just published `@bluejay/access-control@2.6.4` with corrected docs and an example on how to customize the `exists` behavior. See https://github.com/bluebirds-blue-jay/access-control/blob/master/examples/custom-condition-modifier-manager.ts#L12

Let me know if it solves your problem!

**Comment by broom9** [comment]
Thanks! This solved the issue.

-- 
Wei Wei
